working-with-openai/open-ai-api-endpoints.md

```diff
@@ -0,0 +1,156 @@
+| https://api.openai.com/v1/moderations | POST | Moderations | Checks to see if text could be potentially harmful | 
+| https://api.openai.com/v1/models | GET | None | Returns a list of OpenAI models | 
+
+We will focus on using the chat completions endpoint for now, but  each endpoint can be called similarly. Each request body will require some form of input at a minimum and most will require the specific model you would like to use as well. The response you receive will also vary from request to request so make sure you have an understanding of the shape of a response before jumping right in. The full list of OpenAI endpoints along with examples of a request and response body can be found [here](https://platform.openai.com/docs/api-reference/introduction). OpenAI also provides a [model compatability chart](https://platform.openai.com/docs/models/model-endpoint-compatibility) that allows you to see which models work with each endpoint.
+
+## The Chat Completions Endpoint
+
+Before we jump right into integrating the API into a project, let's explore the chat completions endpoint. This particular endpoint is used to mimic and then continue a conversation between the user and an artificial intelligence. In terms of chatGPT, the chat completion endpoint is what is used to interact with the GPT models. The term "chat completion" simply refers to the fact that the response will be the next logical reply to the conversation that is going on. 
+
+### Chat Completion Request Body
+
+As mentioned earlier, the chat completion endpoint is: 
+```
+    POST https://api.openai.com/v1/chat/completions 
+``` 
+Like most POST endpoints, the chat completions endpoint is not complete itself without a request body. Let's go ahead and write a request body and then discuss its individual parts after.
+
+Open up Postman and enter the POST endpoint from above. Make sure you are in your OpenAI environment and then navigate to set up a raw JSON body. In the window, copy and paste the request body below.
+```
+{
+    "model": "gpt-3.5-turbo",
+    "messages": [
+      {
+        "role": "system",
+        "content": "You are a helpful assistant."
+      },
+      {
+        "role": "user",
+        "content": "Hello! Are you able to understand this?"
+      }
+    ]
+}
+```
+
+Now let's look at each piece a little more in depth
+
+| Piece of Code | Notes |
+| ------------- | ----- | 
+| `model` | The model you would like to use for the request. |
+| `messages` | A list of all the messages within a conversation. |
+| `role` | Indicates who the message came from. The most common roles are "user" and "system". |
+| `content` | The actual text of the message |
+
+The four properties above are REQUIRED for your request to work properly. Below are a few other properties that can be included in your request, but the following are all OPTIONAL. 
+
+| Property | Description |
```

Comment:
The first column is splitting the first & last property names over lines, we can set a min-width on that column to prevent it from shrinking too much when it's distracting.

```suggestion
| <div style="min-width:100px;">Property</div>| Description |
```

Reply:
Thank you! I couldn't figure out how to do this quickly, so I skipped it!